src/hooks/useDarkMode.ts

```diff
@@ -0,0 +1,18 @@
+import { useEffect, useState } from 'react';
+
+const isSystemDarkMode = (): boolean => {
+  if (typeof window === 'undefined' || !window.matchMedia) return false;
```

Comment:
As you're not using NextJS, I don't think you need check for the existence of `window`. You might be able to simplify this implementation.